### DISCUSSION TITLE
Supervised in docker su Ubuntu .... non c'è modo di fare il restore delle cfg

### DISCUSSION THREAD

**Discussion by megasauro** [indomus/forum#2501] [HUB: Home Assistant]
Salve,
non so se potete aiutarmi, indicandomi la fattibilità di quanto chiedo o reindirizzandomi ad altro thread già scritto sull'argomento.
Dopo diversi mesi di "test" di HA OS su una VM Virtualbox in ambiente Linux, ho deciso di passare (forse impropriamente, ditemelo voi) alla versione Supervised in docker su Ubuntu.
Sono partito dal fatto che, per crearmi una nuova installazione "pulita" cioè senza tutti i "residui" dei test che avevo fatto nel tempo, ho rifatto la cfg daccapo sulla versione Supervised in docker (2023.05).
Devo dire che la versione che "gira" nel docker, come mi aspettavo è molto veloce, efficace e pratica ... più della VM che avevo prima.
Dopo circa un mese però ho notato che la Supervised in docker non propone e non permette di effettuare aggiornamenti di versione da dentro HA, o almeno questa è quanto ho sperimentato (magari sbagliando), perché dichiara sempre "nessuna aggiornamento disponibile".
Ho allora effettuato un backup delle cfg ed ho proceduto all'aggiornamento del Supervised tramite docker (2023.06).

E qui nascono i problemi: come faccio ad effettuare il restore delle mie cfg e dei miei dati, relativi alla che ho nel backup della 2023.05, nella 2023.06 ??
Se la funzione di restore non è presente all'interno di HA Supervised, non capisco cosa ci faccio con i backup e soprattutto, come posso aggiornare HA ?? 
Esiste già una procedura da seguire per questa operazione ?? Io l'ho cercata, ma senza successo.

Infine, sto forse sbagliando versione di HA e la Supervised non è la versione giusta da impiegare ?

Ogni aiuto è ben accetto.
Grazie in anticipo.
Sauro

**Comment by snakuzzo** [comment]
Ciao.
Direi che bene o male stai sbagliando tutto.
Innanzi tutto, l'installazione di HA Supervised non è ufficialmente supportata su distro diverse da Debian (anche se Debian-like). E' ufficialmente supportata solo su Debian ultima release stable e Debian penultima fino a massimo 4 mesi dall'uscita dell'ultima stable.

Altra precisazione... non è che "HA gira NEL docker". Questa frase non vuol dire niente. Homeassistant, che sia HA OS o HA Supervised su Docker, non è altro che un insieme di container Docker che sono gestiti grazie al Supervisor (anche esso un container docker). L'unica differenza tra le due modalità di installazione è che una (HA OS) è bella e pronta e gira su un OS custom creato ad hoc...l'altra "deve girare" su Debian dopo aver installato Docker più qualche altro prerequisito.

Detto ciò...
- Per quanto riguarda il backup, forse la guida non è aggiornatissima, ma basta perderci un attimoi per capire che puoi ripristinarlo andando su  Settings > System > Backup. Carichi il file cliccando sulla voce apposita cliccando sui tre puntini in alto a destra.

![image](https://github.com/indomus/forum/assets/7205371/36aa30bf-1803-4a0b-b302-1c1a85d7a27c)

Una volta caricato il file lo vedrai tra i backup disponibili. Clicchi sul file e ci sarà la voce "Restore"

![image](https://github.com/indomus/forum/assets/7205371/f6a089e0-5ab4-414b-9cb3-0fd570c79944)


- Gli aggiornamenti, che siano del OS, di Core o degli addon...ti vengono segnalati in automatico 

![image](https://github.com/indomus/forum/assets/7205371/bd30d5ee-9f6a-4274-8961-bb4242eed51e)

Per tutto quello che non va... ricorda... ci sono sempre i log (questi sconosciuti).

Io ti consiglierei di buttare tutto e di rifare l'installazione su Debian Stable

**Comment by megasauro** [comment]
Grazie mille, sapevo che stavo sbagliando qualcosa, speravo non proprio tutto 😮

Il problema "padre" è che nella mia Supervised (sia nella 2023.05 che nella 2023.06) i "tre pallini" per accedere al sottomenù e caricare il backup, proprio non ci sono 😥 :
![image](https://github.com/indomus/forum/assets/99208541/6d6feb21-128f-43b2-a8eb-29191956d32e)

e la pagina degli aggiornamenti dà sempre la medesima visualizzazione, ed il medesimo sottomenù:
![image](https://github.com/indomus/forum/assets/99208541/39371632-ac8f-441f-9b18-13881d03926f)

Detto ciò, accetterei più che volentieri il consiglio di rifare l'installazione su Debian Stable, ma il mio server è un Ubuntu LTS 22.04 ed ho tutta una serie di servizi che non posso (e non vorrei) dover ripristinare ex-novo su una Debian, ripartendo daccapo.

Temo che l'unica alternativa che mi rimane sia di tornare ad installare HA OS su una VM che gira sul mio Ubuntu ... peccato per lo spreco di risorse, a meno che tu non abbia un altra strada da suggerirmi.

Grazie cmq dell'aiuto. 😊

**Comment by megasauro** [comment]
Allora, sicuramente non è un'istallazione canonica su Debian (anche se Ubuntu è un derivato) e quindi l'HA così ottenuto non è "ufficialmente supportato", ma poiché devo dire che HA Supervised su Docker funziona alla grande, impegnando un quantitativo irrisorio di risorse, e visto che non "vedevo" alternative percorribili .... ho cercato di trovare un workaround per riuscire a fare il restore del mio backup (2023.05) sulla nuova versione di HA (2023.06).

Quindi, ho salvato un backup dalla precedente versione e copiato su una cartella diversa da quella dei backup di HA (io uso uno storage dedicato formato da 4 HDD da 2TB, configurati in RAIDZ1 sul SRV Ubuntu, ma non è significativo).
Per evitare che la nuova installazione potesse fare "pasticci", trovando la precedente versione installata, ho fermato il container di HA, ho eliminato il container (vecchia versione di HA) ed ho anche eliminato i file (tanto avevo il backup salvato altrove) del container di HA .... insomma "ho pulito il tavolo", se la frase può aver senso.
Poi, seguendo le indicazioni delle guide di inDomus (https://indomus.it/formazione/utilita-come-aggiornare-home-assistant-core/#dockerlinux), ho installato il container della nuova versione di HA (last release).

A questo punto, prima di connettermi ad HA ho fermato il container, ed ho copiato i file del backup della 2023.05 (che avevo espanso in una cartella, visto che HA li crea già compressi in tar.gz), nella cartella del nuovo HA 2023.06 sovrascrivendo quelli dell'installazione del 2023.06 (che in realtà erano di fatto "privi di contenuto", se non per le cfg di default).

Riavviato il container, questo è attivo e funzionante (lo verifico dall'interfaccia web di Portainer, che ho installato perché il mio SRV è headless, e così posso gestire i container da GUI web):
![image](https://github.com/indomus/forum/assets/99208541/1c7ceb98-cbc9-4ab4-88a1-33ce0156c5f3)

Rilanciato HA, e loggandomi con le medesime credenziali che avevo impostato sulla vecchia 2023.05, ho verificato che la versione sia la 2023.06 
![image](https://github.com/indomus/forum/assets/99208541/fa9c5068-b7a7-482b-9c71-3d6afe70a1a6)
e che il mio HA 2023.06 abbia le plance configurate come il vecchio 2023.05
![image](https://github.com/indomus/forum/assets/99208541/ed6df2a9-455a-42e5-96aa-f08ab5d9f9d9)

![image](https://github.com/indomus/forum/assets/99208541/161b5de9-1926-4ddf-a75b-cb3079d7dff6)
(i dati del consumo energetico sono "fermi" al 09 giugno perché stupidamente ho usato il backup del 09 per il restore .... e non quello di oggi, ma non è un grosso problema).
Così sembra che stia funzionando senza difficoltà ... vediamo nei prossimi giorni come si comporta.

Ripeto non è la soluzione canonica da seguire, ma per chi sta "sperimentando" HA Supervised su Docker e si trova davanti al mio stesso problema .... potrebbe essere una soluzione percorribile.

**Reply by snakuzzo** [reply]
Non c'era bisogno di specificare tutta sta roba, con tanto di screenshot.
Hai semplicemente installato core su docker (che resta l'installazione più indicata...sempre...se hai almeno un minimo di dimestichezza con Docker). Tu però prima parlavi di HA Supervised, che non è solo Core, ma c'è anche tutto il resto, così come su HA OS. E nello specifico parlo anche degli addon.
